### DISCUSSION TITLE
Sync newly added Variables

### DISCUSSION THREAD

**Discussion by efraim14** [envault/envault#25] [Q&A] [answered]
Hello,

just recently discovered this great tool for my projects. I am currently running several repos on different servers and really need a centralized environment variables for all the repos, so i can manage to sync .env files without having to ssh into the server.

My questions are : 
1. Is it possible to sync not only variable that exist in .env file, but also sync new variable that added on envault into .env?
2. Does the unique token on `npx envault <domain> <apps_id> <unique_token>` always active? Is it possible to run using the same command forever on my pipelines to sync into envault?

**Comment by danharrin** [comment, marked as answer]
Hey @efraim14! Thanks for your interest in the project.

You are now able to pull newly created variables using `npx envault --constructive`.

Regarding your other question... yes it is possible. Once you run `npx envault <domain> <app> <token>` once, a permanent authentication token is created and stored in your `.envault.json` file for all future requests. If your pipeline is able to access this file, it can simply run `npx envault` each time to sync changes. Please let me know if this is clear or if there is an improvement we can make to this process!
